slamd/discovery/processing/experiment/experiment_conductor.py

```diff
@@ -0,0 +1,167 @@
+# Adapted from the original Sequential Learning App
+# https://github.com/BAMresearch/SequentialLearningApp
+import pandas as pd
+from scipy.spatial import distance_matrix
+from sklearn.gaussian_process import GaussianProcessRegressor
+from sklearn.gaussian_process.kernels import RBF, ConstantKernel
+
+from slamd.common.error_handling import ValueNotSupportedException
+from slamd.discovery.processing.experiment.experiment_postprocessor import ExperimentPostprocessor
+from slamd.discovery.processing.experiment.experiment_preprocessor import ExperimentPreprocessor
+from slamd.discovery.processing.experiment.slamd_random_forest import SlamdRandomForest
+from slamd.discovery.processing.experiment.experiment_model import ExperimentModel
+
+
+# Attention - suppressing expected Gaussian Regressor warnings
+import warnings
+from sklearn.exceptions import ConvergenceWarning
+warnings.filterwarnings("ignore", category=ConvergenceWarning)
+
+
+class ExperimentConductor:
+
+    @classmethod
+    def run(cls, exp):
+        ExperimentPreprocessor.preprocess(exp)
+        cls._fit_model_and_predict(exp)
+        cls._calculate_utility(exp)
+        cls._calculate_novelty(exp)
+
+        return ExperimentPostprocessor.postprocess(exp)
+
+    @classmethod
+    def _fit_model_and_predict(cls, exp):
+        if exp.model == ExperimentModel.RANDOM_FOREST.value:
+            regressor = SlamdRandomForest()
+        elif exp.model == ExperimentModel.GAUSSIAN_PROCESS.value:
+            # Hyperparameters from previous implementation of the app (jupyter notebook)
+            kernel = ConstantKernel(1.0, (1e-3, 1e3)) * RBF(10, (1e-2, 1e2))
+            regressor = GaussianProcessRegressor(kernel=kernel, n_restarts_optimizer=9, random_state=42)
+        else:
+            raise ValueNotSupportedException(message=f'Invalid model: {exp.model}')
```

Comment:
This is correct, but I did not want to use a simple if... else in case more models are added in the future, and I did not want to use an if... elif without an else clause. I adjusted the error message to reflect that it should never occur.